lib/uring/uring_stubs.c

```diff
@@ -184,6 +185,40 @@ ocaml_uring_submit_nop(value v_uring, value v_id) {
   CAMLreturn(Val_true);
 }
 
+value ocaml_uring_submit_timeout(value v_uring, value v_id, value v_clock, value v_timeout)
+{
+  CAMLparam4(v_uring, v_id, v_clock, v_timeout);
+  struct __kernel_timespec t;
+  struct io_uring* ring = Ring_val(v_uring);
+  struct io_uring_sqe* sqe;
+  int res, clock;
+
+  if (v_clock == caml_hash_variant("Clock_mono"))
+    clock = IORING_TIMEOUT_BOOTTIME;
+  else if (v_clock == caml_hash_variant("Clock_sys"))
+    clock = IORING_TIMEOUT_REALTIME;
+  else
+    clock = 0;
+
+  t.tv_sec = 0;
+  t.tv_nsec = Int64_val(v_timeout);
+
+  sqe = io_uring_get_sqe(ring);
+  if(!sqe) {
+    res = io_uring_submit(ring);
+    if (res < 0)
+      CAMLreturn(Val_false);
+
+    sqe = io_uring_get_sqe(ring);
+    if(!sqe)
+      CAMLreturn(Val_false);
+  }
+  io_uring_prep_timeout(sqe, &t, 0, clock);
+  io_uring_sqe_set_data(sqe, (void *)Long_val(v_id));
+  sqe->user_data = LIBURING_UDATA_TIMEOUT;
```

Comment:
What is this for?

Reply:
Do you mean `sqe->user_data = LIBURING_UDATA_TIMEOUT;` ?

It follows https://github.com/axboe/liburing/blob/02adcf497c9f4ca20175b291f8269faff4da958c/src/queue.c#L278